pkg/apply/solver/solver.go

```diff
@@ -86,74 +87,139 @@ func (tq *TaskQueue) ToActionGroups() []event.ActionGroup {
-	klog.V(2).Infoln("adding delete inventory task")
-	t.tasks = append(t.tasks, &task.DeleteInvTask{
-		TaskName:  fmt.Sprintf("delete-inventory-%d", t.deleteInvCounter),
-		InvClient: t.InvClient,
-		InvInfo:   inv,
-		DryRun:    dryRun,
-	})
-	t.deleteInvCounter++
-	return t
+// Build returns the queue of tasks that have been created
+func (t *TaskQueueBuilder) Build(o Options) *TaskQueue {
+	var tasks []taskrunner.Task
+
+	// reset counters
+	t.applyCounter = 0
+	t.pruneCounter = 0
+	t.waitCounter = 0
+
+	applyObjs := t.Collector.FilterInvalidObjects(t.applyObjs)
+	pruneObjs := t.Collector.FilterInvalidObjects(t.pruneObjs)
+
+	if len(applyObjs) > 0 {
+		klog.V(2).Infoln("adding inventory add task (%d objects)", len(applyObjs))
+		tasks = append(tasks, &task.InvAddTask{
+			TaskName:  "inventory-add-0",
+			InvClient: t.InvClient,
+			InvInfo:   t.invInfo,
+			Objects:   applyObjs,
+			DryRun:    o.DryRunStrategy,
+		})
+	}
+
+	// Use the "depends-on" annotation to create a graph, ands sort the
+	// objects to apply into sets using a topological sort.
+	if len(applyObjs) > 0 {
+		applySets, err := graph.SortObjs(applyObjs)
+		if err != nil {
+			t.Collector.Collect(err)
+		}
+		for _, applySet := range applySets {
+			applySet = t.Collector.FilterInvalidObjects(applySet)
+			if len(applySet) == 0 {
+				continue
+			}
+			tasks = append(tasks,
+				t.newApplyTask(applySet, t.ApplyFilters, t.ApplyMutators, o))
+			// dry-run skips wait tasks
+			if !o.DryRunStrategy.ClientOrServerDryRun() {
+				applyIds := object.UnstructuredSetToObjMetadataSet(applySet)
+				tasks = append(tasks,
+					t.newWaitTask(applyIds, taskrunner.AllCurrent, o.ReconcileTimeout))
+			}
+		}
+	}
+
+	if o.Prune && len(pruneObjs) > 0 {
+		// Use the "depends-on" annotation to create a graph, ands sort the
+		// objects to prune into sets using a (reverse) topological sort.
+		pruneSets, err := graph.ReverseSortObjs(pruneObjs)
+		if err != nil {
+			t.Collector.Collect(err)
+		}
+		for _, pruneSet := range pruneSets {
+			pruneSet = t.Collector.FilterInvalidObjects(pruneSet)
+			if len(pruneSet) == 0 {
+				continue
+			}
+			tasks = append(tasks,
+				t.newPruneTask(pruneSet, t.PruneFilters, o))
+			// dry-run skips wait tasks
+			if !o.DryRunStrategy.ClientOrServerDryRun() {
+				pruneIds := object.UnstructuredSetToObjMetadataSet(pruneSet)
+				tasks = append(tasks,
+					t.newWaitTask(pruneIds, taskrunner.AllNotFound, o.PruneTimeout))
+			}
+		}
+	}
+
+	// TODO: add InvSetTask when destroying to retain undeleted objects
+	if !o.Destroy {
+		klog.V(2).Infoln("adding inventory set task")
+		prevInvIds, _ := t.InvClient.GetClusterObjs(t.invInfo)
+		tasks = append(tasks, &task.InvSetTask{
+			TaskName:      "inventory-set-0",
+			InvClient:     t.InvClient,
+			InvInfo:       t.invInfo,
+			PrevInventory: prevInvIds,
+			DryRun:        o.DryRunStrategy,
+		})
+	}
+
+	if o.Destroy {
```

Comment:
You can change this to 
` else {`

Reply:
I could, but the above Destroy check is temporary. In the near future I want to always run InvSetTask so that the inventory gets updated if any objects were NOT deleted.

Reply:
Ack

Reply:
done